### PR TITLE
[WIP] Update: Add Perl dependencies needed for GEO converter

### PR DESCRIPTION
**Pull request by anjaf**
This adds missing Perl installs that are needed to run the GEO to MAGE-TAB conversion functionality. 

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by dpryan79** [comment]
You'll need to remove this from the build-fail-blacklist. It wasn't building during the last perl migration. In general we try to avoid using cpan within recipes, since the results don't always play well with other packages.

File: recipes/perl-atlas-modules/build.sh [hunk 1]
```diff
@@ -26,23 +26,21 @@ fi
...
-                    URL::Encode \
                     Config::YAML \
                     File::Basename \
                     Bio::MAGETAB \
                     Date::Parse \
                     Test::MockObject \
                     Text::TabularDisplay \
-                    Log::Dispatch::File \
-                    IO::CaptureOutput \
                     TIMB/DBI-1.636.tar.gz \
```
Comment (pcm32) [inline recipes/perl-atlas-modules/build.sh:35]: DBI already gets installed by conda:
```
2022-03-22T11:46:41.2093226Z 11:46:41 BIOCONDA INFO (OUT)     perl-dbi:                        1.643-pl5321hec16e2b_1        bioconda
```
I would try removing this.

File: recipes/perl-atlas-modules/build.sh [hunk 2]
```diff
@@ -26,23 +26,21 @@ fi
...
                     MIYAGAWA/Class-Trigger-0.15.tar.gz \
                     TMTM/Class-DBI-v3.0.17.tar.gz \
                     DETI/Proc/Proc-Daemon-0.14.tar.gz \
```
Comment (pcm32) [inline recipes/perl-atlas-modules/build.sh:38]: I would try un pining these and possibly move them to `package::name` usage (to un pin them).

**Comment by anjaf** [comment]
We found an alternative solution without modifying the existing package. Closing this now.

**Pull request closed without merging**